tests/src/Kernel/Entity/OgMembershipReferenceItemListTest.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\system\Tests\Entity\OgMembershipReferenceItemTest.
+ */
+
+namespace Drupal\Tests\og\Kernel\Entity;
+
+use Drupal\Component\Utility\Unicode;
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\entity_test\Entity\EntityTest;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\og\Entity\OgMembership;
+use Drupal\og\Og;
+use Drupal\og\OgMembershipInterface;
+
+/**
+ * Tests OgMembershipReferenceItem and OgMembershipReferenceItemList classes.
+ *
+ * @group Entity
+ * @coversDefaultClass \Drupal\og\Plugin\Field\FieldType\OgMembershipReferenceItemList
+ */
+class OgMembershipReferenceItemListTest extends KernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = ['entity_test', 'user', 'field', 'entity_reference', 'og', 'system'];
+
+  /**
+   * Array with the bundle IDs.
+   *
+   * @var array
+   */
+  protected $bundles;
+  protected $fieldName;
+  protected $groups;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Add membership and config schema.
+    $this->installConfig(['og']);
+    $this->installEntitySchema('entity_test');
+    $this->installEntitySchema('og_membership');
+    $this->installEntitySchema('user');
+
+    // Create several bundles.
+    for ($i = 0; $i <= 4; $i++) {
+      $bundle = EntityTest::create([
+        'type' => Unicode::strtolower($this->randomMachineName()),
+        'name' => $this->randomString(),
+      ]);
+
+      $bundle->save();
+      $this->bundles[] = $bundle->id();
+    }
+    for ($i = 0 ; $i < 2; $i++) {
+      $bundle = $this->bundles[$i];
+      Og::groupManager()->addGroup('entity_test', $bundle);
+      $group = EntityTest::create(['type' => $bundle]);
+      $group->save();
+      $this->groups[] = $group;
+    }
+    $this->fieldName = strtolower($this->randomMachineName());
+
+    FieldStorageConfig::create([
+      'field_name' => $this->fieldName,
+      'type' => 'og_membership_reference',
+      'entity_type' => 'entity_test',
+      'cardinality' => -1,
+      'settings' => [
+        'target_type' => 'entity_test',
+      ],
+    ])->save();
+    FieldConfig::create([
+      'field_name' => $this->fieldName,
+      'entity_type' => 'entity_test',
+      'bundle' => $this->bundles[2],
+      'label' => $this->randomString(),
+      'settings' => [
+        'handler' => 'default',
+        'handler_settings' => [
+          'target_bundles' => [$this->bundles[0], $this->bundles[1]],
+        ],
+      ],
+    ])->save();
+  }
+
+  /**
+   * Test creating and saving og membership reference field items.
+   */
+  public function testMembershipSave() {
+    $run_query = function ($id) {
```

Comment:
is it common to have those functions inside the tests? In other words, why not a method?